### PR TITLE
Shopping Cart: Clean up lib/cart-values and remove unused functions

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

In https://github.com/Automattic/wp-calypso/pull/51105 we removed a bunch of unused code from lib/cart-values. This PR does a little refactoring to remove some more. It does the following:

- Remove `addCartItem`. It was only used in signup and adding products to a cart manually is trivial.
- Remove `getAllCartItemsSorted`. It was only used for the popover cart and now that the shopping-cart endpoint does sorting it's not really needed.
- Remove `sortProducts` (from calypso-products). This function was originally for the cart, but now it was only used by the purchases selector. This seems unnecessary.
- Remove `preprocessCartForServer`. It was only used in signup and didn't actually do anything but pull in the cart and spit it out.
- Remove types from lib/cart-values. They were already deprecated in favor of the types in `@automattic/shopping-cart`.
- Remove lodash from lib/cart-values. All the uses were easily handled by native JS.

#### Testing instructions

Automated tests should be sufficient.

Also verify that the removed functions mentioned above are not used anywhere.

### CODE REVIEW

File: client/lib/cart-values/types.ts [hunk 1]
```diff
@@ -1,16 +0,0 @@
...
-	ResponseCart,
-	ResponseCartProduct,
-	ResponseCartProductExtra,
-} from '@automattic/shopping-cart';
-
-// These types are deprecated. Please use types from @automattic/shopping-cart directly.
-
-export type CartItemValue = ResponseCartProduct;
-
-export type CartItemExtra = ResponseCartProductExtra;
```
Comment (michaeldcain) [inline client/lib/cart-values/types.ts:14]: Should [this](https://github.com/Automattic/wp-calypso/blob/remove/unused-cart-values-functions/client/lib/cart-values/cart-items.js#L678) be updated?

File: client/lib/cart-values/types.ts [hunk 2]
```diff
@@ -1,16 +0,0 @@
...
-
-export type CartValue = ResponseCart;
```
Comment (michaeldcain) [inline client/lib/cart-values/types.ts:16]: There are a number of "cart as `CartValue` object" comments in `/lib/analytics/`. Should they be updated to `ResponseCart`?

File: client/lib/cart-values/cart-items.js [hunk 3]
```diff
@@ -54,97 +46,20 @@ import {
...
-/**
- * Determines if the given cart item should replace the cart.
- * This can happen if the given item:
- * - will result in mixed renewals/non-renewals or multiple renewals (excluding privacy protection).
- *
- * @param {CartItemValue} cartItem - `CartItemValue` object
- * @param {object} cart - the existing shopping cart
- * @returns {boolean} whether or not the item should replace the cart
- */
-export function cartItemShouldReplaceCart( cartItem, cart ) {
```
Comment (michaeldcain) [inline client/lib/cart-values/cart-items.js:98]: There's a [comment reference to this function here](https://github.com/Automattic/wp-calypso/blob/remove%2Funused-cart-values-functions/packages/shopping-cart/src/cart-functions.ts#L206).

**Review by michaeldcain — APPROVED** [review]
There are some comments referencing the removed functions and types, but otherwise this LGTM

File: client/lib/cart-values/types.ts (hunk 1 above)
Comment (sirbrillig) [reply client/lib/cart-values/types.ts:14]: Yes. Good catch.

File: client/lib/cart-values/types.ts (hunk 2 above)
Comment (sirbrillig) [reply client/lib/cart-values/types.ts:16]: 👍  Thanks!

File: client/lib/cart-values/cart-items.js (hunk 3 above)
Comment (sirbrillig) [reply client/lib/cart-values/cart-items.js:98]: That's a good idea. I updated the comment.

**Pull request merged**